src/pet.js

```diff
@@ -0,0 +1,56 @@
+function Pet(name) {
+    this.name = name;
+    this.age = 0;
+    this.hunger = 0;
+    this.fitness = 10;
+    this.children = [];
+};
+
+Pet.prototype = {
+    get isAlive() {
+        return this.age < 30 && this.hunger < 10 && this.fitness > 0;
+    }
+  };
+
+Pet.prototype.growUp = function() {
+    if (!this.isAlive) throw new Error("Your pet is no longer alive :(");
+    this.age += 1;
+    this.hunger += 5;
+    this.fitness -= 3;
+};
+
+Pet.prototype.walk = function() {
+    const fitnessIncrementPerWalk = 4;
+    const maximumFitnessLevel = 10;
+    if (!this.isAlive) throw new Error("Your pet is no longer alive :(");
+    this.fitness <= 6 ? this.fitness += fitnessIncrementPerWalk : this.fitness += (maximumFitnessLevel - this.fitness);
+};
+
+Pet.prototype.feed = function() {
+    const hungerDecrementPerFeed = 3;
+    if (!this.isAlive) throw new Error("Your pet is no longer alive :(");
+    this.hunger >= 3 ? this.hunger -= hungerDecrementPerFeed : this.hunger -= this.hunger;
+};
+
+Pet.prototype.checkUp = function() {
+    const minimumDesiredFitness = 3;
+    const maximumDesiredHunger = 5;
+    return !this.isAlive ? "Your pet is no longer alive :(" :
+    this.fitness <= minimumDesiredFitness && this.hunger < maximumDesiredHunger ? "I need a walk":
+    this.hunger >= maximumDesiredHunger && this.fitness > minimumDesiredFitness ? "I am hungry":
+    this.hunger >= maximumDesiredHunger && this.fitness <= minimumDesiredFitness ? "I am hungry AND I need a walk": 
+    "I feel great!";
```

Comment:
Shouldn't chain ternary operators like this. If you have more than 1 condition you are checking for, you should write separate if statements instead as it makes it more readable. i.e.

```
if (!this.IsAlive) {
  return "Your pet is no longer alive :(";
}

if (this.fitness <= minimunDesiredFitness && this.hunger >= maximumDesiredHunger) {
  return "I am hungry AND I need a walk";
}

if (this.fitness <= minimumDesiredFitness) {
  return "I need a walk";
}

if (this.hunger >= maximimDesiredHunger) {
  return "I am hungry";
}

return "I feel great!";
```

note: as I am checking the `"I am hungry AND I need a walk"` statement first, I don't need to check the hunger state in the `"I need a walk"` statement or the fitness state in the `"I am hungry"` statement.

this code still has a lot of repetition, to reduce this you could rewrite the checkUp function as follows

```
pet.prototype.checkUp = function() {
  if (!this.IsAlive) return "Your pet is no longer alive :(";

  const minimumDesiredFitness = 3;
  const checkFitness = this.fitness <= minimumDesiredFitness;
  const maximumDesiredHunger = 5;
  const checkHunger = this.hunger >= maximumDesiredHunger;
  const hungryString = "I am hungry";
  const walkString = "I need a walk";

  if (checkFitness && checkHunger) return `${hungryString} AND ${walkString}`;
  if (checkFitness) return walkString;
  if (checkHunger) return hungryString;
  return "I feel great!";
}

```